src/20250814_Multi_GHOInkLaunch/abstraction/AaveV3GHOLaunch.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IACLManager} from 'aave-address-book/AaveV3.sol';
+import {ITokenAdminRegistry} from 'src/interfaces/ccip/ITokenAdminRegistry.sol';
+import {IGhoBucketSteward} from 'src/interfaces/IGhoBucketSteward.sol';
+import {IGhoToken} from 'src/interfaces/IGhoToken.sol';
+import {AaveV3GHOLane} from './AaveV3GHOLane.sol';
+import {GhoCCIPChains} from './constants/GhoCCIPChains.sol';
+
+/**
+ * @title GHO Launch
+ * @author Aave Labs
+ * @notice Proposal Executor for Aave V3 GHO Launch
+ */
+abstract contract AaveV3GHOLaunch is AaveV3GHOLane {
+  IGhoToken public immutable GHO_TOKEN;
+  ITokenAdminRegistry public immutable TOKEN_ADMIN_REGISTRY;
+  IACLManager public immutable ACL_MANAGER;
+  address public immutable GHO_BUCKET_STEWARD;
+  address public immutable GHO_AAVE_STEWARD;
+  address public immutable GHO_CCIP_STEWARD;
+  address public immutable OWNER;
+
+  uint128 internal constant DEFAULT_CCIP_BUCKET_CAPACITY = 40_000_000e18;
+
+  constructor(GhoCCIPChains.ChainInfo memory localChainInfo) AaveV3GHOLane(localChainInfo) {
+    GHO_TOKEN = IGhoToken(localChainInfo.ghoToken);
+    GHO_BUCKET_STEWARD = localChainInfo.ghoBucketSteward;
+    GHO_AAVE_STEWARD = localChainInfo.ghoAaveCoreSteward;
+    GHO_CCIP_STEWARD = localChainInfo.ghoCCIPSteward;
+    ACL_MANAGER = IACLManager(localChainInfo.aclManager);
+    TOKEN_ADMIN_REGISTRY = ITokenAdminRegistry(localChainInfo.tokenAdminRegistry);
+    OWNER = localChainInfo.owner;
+  }
+
+  function execute() external override {
+    _acceptTokenAdminRegistryAdminRole();
+    _acceptTokenPoolOwnership();
+    _setupStewardsAndTokenPoolOnGho();
+    _applyChainUpdatesInTokenPool();
+    _setPoolInTokenAdminRegistry();
+  }
+
+  function getCCIPBucketCapacity() public pure virtual returns (uint128) {
```

Comment:
all these methods i think need validation that addresses and values are correct (not 0, not address(0) etc) If there is a case where some address or value needs to be 0, then it should get overriden in child (with comment explaining why)

Reply:
Do you mean in tests or where? For example this one has already a non-zero default value, and if wants to be changed it will need to be overridden in the child. And given that the setup of the child with current abstraction is minimal, it would be easy to review each overridden behavior/constant